frontend/simple/views/CreateGroup.vue

```diff
@@ -236,22 +131,82 @@ export default {
       } catch (error) {
         console.error(error)
         this.errorMsg = L('Failed to Create Group')
+        return
+      }
+
+      try {
+        this.errorMsg = null
+        // TODO: as invitees are successfully invited display in a
+        // seperate invitees grid and add them to some validation for duplicate invites
+        for (let invitee of this.form.invitees) {
+          // We need to have the latest mailbox attribute for the user
+          const mailbox = await backend.latestHash(invitee.state.attributes.mailbox)
+          const sentDate = new Date().toString()
+
+          // We need to post the invite to the users' mailbox contract
+          const invite = new Events.HashableMailboxPostMessage(
+            {
+              from: this.$store.getters.currentGroupState.groupName,
+              headers: [this.$store.state.currentGroupId],
+              messageType: Events.HashableMailboxPostMessage.TypeInvite,
+              sentDate
+            },
+            mailbox
+          )
+          await backend.publishLogEntry(invitee.state.attributes.mailbox, invite)
+
+          // We need to make a record of the invitation in the group's contract
+          const latest = await backend.latestHash(this.$store.state.currentGroupId)
+          const invited = new Events.HashableGroupRecordInvitation(
+            {
+              username: invitee.state.attributes.name,
+              inviteHash: invite.toHash(),
+              sentDate
+            },
+            latest
+          )
+          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+        }
+      } catch (error) {
+        console.error(error)
+        // TODO: Create More descriptive errors
+        this.errorMsg = L('Failed to Invite Users')
```

Comment:
This seems like a strange place to put this code...

It would be better, if possible, to keep components as self-contained as possible, and therefore keep the invite code with the invite component, and the group creation code with the group creation component.

Reply:
with the new workflow, the invitee component is only for gathering the users to be invited once the group is created, and the actual invitation only takes place once all group data is in order and the finish button is clicked, together with group creation. there is also no group to invite people to when we interact with the invitee component. I would definitely want to avoid putting the invitation itself back together with the invitee collection. but I can move this to its own component if you think that is necessary.

Reply:
OK. I guess this is fine, although we should still at some point in the future think through all the potential error situations here... It feels to me like something might break if something goes wrong. For now I do not see anything obviously critical at least.